File: Server/src/main/content/global/skill/support/slayer/npc/CaveBugNPC.java
package content.global.skill.support.slayer.npc;

import core.game.node.entity.Entity;
import core.game.node.entity.npc.AbstractNPC;
import core.game.node.entity.player.Player;
import core.game.world.map.Location;
import core.plugin.Initializable;

/**
 * The Cave bug npc.
 */
@Initializable
public class CaveBugNPC extends AbstractNPC {


//	private static final ZoneBorders CAVE_BORDER = new ZoneBorders(3139, 9534, 3260, 9587);

    /**
     * Instantiates a new Cave bug npc.
     */
    public CaveBugNPC() {
        super(-1, null);
    }

    /**
     * Instantiates a new Cave bug npc.
     *
     * @param id       the id
     * @param location the location
     */
    public CaveBugNPC(int id, Location location) {
        super(id, location);
    }

    @Override
    public void finalizeDeath(Entity killer) {
        super.finalizeDeath(killer);
        if (killer instanceof Player) {
            Player p = killer.asPlayer();
        }
    }

    @Override
    public AbstractNPC construct(int id, Location location, Object... objects) {
        return new CaveBugNPC(id, location);
    }

    @Override
    public int[] getIds() {
        return new int[]{1832, 1833, 5750};
    }

}

File: Server/src/main/content/global/skill/support/slayer/npc/CaveBugNPC.kt
package content.global.skill.support.slayer.npc

import cfg.consts.NPCs
import core.game.node.entity.Entity
import core.game.node.entity.npc.AbstractNPC
import core.game.node.entity.player.Player
import core.game.world.map.Location
import core.plugin.Initializable

/**
 * Cave bug NPC.
 */
@Initializable
class CaveBugNPC(id: Int = 0, location: Location? = null) : AbstractNPC(id, location) {

    override fun finalizeDeath(killer: Entity) {
        super.finalizeDeath(killer)
        if (killer is Player) {
            val p = killer.asPlayer()
        }
    }

    override fun construct(id: Int, location: Location, vararg objects: Any): AbstractNPC {
        return CaveBugNPC(id, location)
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.CAVE_BUG_1832, NPCs.CAVE_BUG_LARVA_1833, NPCs.CAVE_BUG_5750)
    }
}
